app/src/main/java/com/example/githubclone/presentation/common/Common.kt

```diff
@@ -0,0 +1,83 @@
+package com.example.githubclone.presentation.common
+
+import android.app.Activity
+import android.util.Log
+import android.widget.Toast
+import com.google.android.gms.tasks.Task
+import com.google.firebase.auth.*
+
+fun signIn(
+    activity: Activity,
+    navigateToHomeScreen: () -> Unit,
+) {
+
+    val firebaseAuth: FirebaseAuth = FirebaseAuth.getInstance()
+    val provider: OAuthProvider.Builder = OAuthProvider.newBuilder("github.com")
+    val pendingResultTask: Task<AuthResult>? = firebaseAuth.pendingAuthResult
+
+    if (pendingResultTask != null) {
```

Comment:
Might not always be the case, but using `nullableVariable?.let { nonNullVariable -> ... } ?: { ... }` is generally safer than using the `if-else` block for checking nulls.

Reply:
done, thank you for the info